File: engine/model/src/main/scala/tools/packet/playerinteraction/TradeItemAdd.scala
package tools.packet.playerinteraction

import client.inventory.Item
import net.opcodes.SendOpcode
import tools.packet.PacketInput

class TradeItemAdd( private var _number: Byte,  private var _item: Item) extends PacketInput {
     def number: Byte = _number
     def item: Item = _item

  override def opcode(): SendOpcode = SendOpcode.PLAYER_INTERACTION
}

File: engine/engine-model/src/main/java/tools/packet/character/interaction/TradeItemAdd.java
package tools.packet.character.interaction;

import client.inventory.Item;
import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record TradeItemAdd(Byte number, Item item) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.PLAYER_INTERACTION;
   }
}